app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -198,37 +209,73 @@ class ConversationViewModel @Inject constructor(
         }
     }
 
+    fun downloadAsset(messageId: String) {
+        viewModelScope.launch {
+            if (!isAssetDownloaded(messageId))
+                updateAssetMessageDownloadStatus(IN_PROGRESS, conversationId, messageId)
+
+            val result = getRawAssetData(conversationId, messageId)
```

Comment:
perhaps instead of getting the asset from the list of messages inside getAssetName() or inside IsassetDownloaded() each time we could do it once here by using : 

```
conversationViewState.messages.firstOrNull {
        it.messageHeader.messageId == messageId && it.messageContent is AssetMessage
    }
```

and remove this responsbility from those functions